File: web/src/api/bom.ts
import { apiClient } from './client'

export type InspectScope = 'pending' | 'inspected' | 'skipped'

export interface BomItem {
  id: string
  project_id: string
  designator: string
  value: string | null
  package: string | null
  qty: number | null
  position_hint: string | null
  inspect_scope: InspectScope
  mi_likely: boolean | null
  component_type: string | null
  defect_history_count: number
  extra: Record<string, unknown> | null
}

interface Envelope<T> {
  status: boolean
  message: string
  data: T
}

export async function listBomItems(projectId: string): Promise<BomItem[]> {
  const { data } = await apiClient.get<Envelope<BomItem[]>>(
    `/projects/${projectId}/bom_items`,
  )
  return data.data
}

File: src/indusia_visual_editor/schemas/bom.py
"""Pydantic schemas for the BOM items API."""

import uuid

from pydantic import BaseModel, ConfigDict

from indusia_visual_editor.db.models import InspectScope


class BomItemRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    project_id: uuid.UUID
    designator: str
    value: str | None
    package: str | None
    qty: int | None
    position_hint: str | None
    inspect_scope: InspectScope
    mi_likely: bool | None
    component_type: str | None
    defect_history_count: int
    extra: dict | None
